src/lib/helpers.ts

```diff
@@ -69,59 +70,45 @@ export const mergeSimilarItems =
   }
 
 export const handleScrollToSelectedTarget = (view: EditorView): boolean => {
-  const { tr, selection } = view.state
-
-  // Get the nodes at the selection's start and end positions
-  const nodeAtFrom = tr.doc.nodeAt(selection.$from.pos)
-  const nodeAtTo = tr.doc.nodeAt(selection.$to.pos)
-  // Get the DOM element at the selection's position
-  const domAtSelectionFrom = view.domAtPos(selection.$from.pos)
-    .node as HTMLElement
-
-  if (!nodeAtFrom) {
+  const selection = view.state.selection
+  if (selection.empty) {
     return false
   }
 
-  // Determine the target element to scroll to
-  const targetElement =
-    nodeAtFrom.type === schema.nodes.bibliography_item
-      ? (document.getElementById(nodeAtFrom.attrs.id) as HTMLElement)
-      : nodeAtTo?.type === schema.nodes.highlight_marker
-      ? (document.getElementById(nodeAtTo?.attrs.id) as HTMLElement)
-      : (document.getElementById(nodeAtFrom.attrs.id) as HTMLElement)
+  const node = view.state.doc.nodeAt(selection.from)
+  if (!node) {
+    return false
+  }
 
-  // Fallback to the DOM element at the selection's position
-  const scrollTarget = targetElement || domAtSelectionFrom
+  let target: Element | null
 
-  if (!scrollTarget) {
-    return false
+  if (node.attrs.id) {
+    target = view.dom.querySelector(`[id="${node.attrs.id}"]`)
+  } else {
+    target = view.domAtPos(selection.from).node as Element
```

Comment:
would it make sense just to do view.domAtPos in all cases even if there is an id in attributes?

Reply:
The id part is needed when the node display is not managed by the editor, namely bibliography_item. The domAtPos part was added later as PM doesn't scroll properly when in view mode.